File: src/cleansia_android/core/src/main/java/cz/cleansia/core/ui/state/ActionState.kt
package cz.cleansia.core.ui.state

/**
 * Three-state machine for one-shot UI actions — the `:core` home of the type so both apps share one
 * definition.
 *
 * The customer app still carries an identical copy; **they must stay in step until that one is
 * removed.**
 */
sealed interface ActionState {
    /** No action in flight, no inline error to surface. The default + post-success state. */
    data object Idle : ActionState

    /** Action is in flight — drives spinners + disables submit buttons. */
    data object Submitting : ActionState

    /** Action failed. [message] is pre-localized — call sites render it directly. */
    data class Error(val message: String) : ActionState
}

File: src/cleansia_ios/CleansiaCore/Sources/CleansiaCore/State/ActionState.swift
import Foundation

public enum ActionState: Equatable {
    case idle
    case submitting
    case error(String)
}

public extension ActionState {
    var isSubmitting: Bool {
        if case .submitting = self { return true }
        return false
    }

    var errorMessage: String? {
        if case let .error(message) = self { return message }
        return nil
    }
}
